thevoid/signal_handler.cpp

```diff
@@ -0,0 +1,139 @@
+#include "server.hpp"
+
+#include <mutex>
+#include <set>
+#include <thread>
+#include <atomic>
```

Comment:
It does not compile with gcc 4.4. You should include either atomic or cstdatomic header depends on compiler.